### PR TITLE
First pass at an implementation of the policystore client

### PR DESCRIPTION
**Pull request by smulube**
**Work in progress pull request.**

The policystore is just responsible for returning a list of availablepolicies. It doesn't take any parameters and just returns an array containing the returned policies.

Opening this PR just to show a bit of work I was doing this afternoon with @jordibuj's help, incase anyone has any comments.

The axios testing was a little unintuitive and hard to get working, and I'm not entirely convinced I have defined them properly.

### CODE REVIEW

File: src/api/policystore-client.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  constructor(url = 'https://policystore.decodeproject.eu') {
+    this.url = url;
+  }
+
+  async listPolicies() {
+    try {
+      const url = `${this.url}${prefix}/ListEntitlementPolicies`;
+      debugLog('Going to call: ', url);
+
+      const { policies } = await axios.post(url, {}, { headers: { 'Content-Type': 'application/json' } });
```
Comment (smulube) [inline src/api/policystore-client.js]: need to understand what happens when axios gets back an error http response - does this go down the error handling route, or do we need to handle that in this clause?

Comment (jordibuj) [reply src/api/policystore-client.js]: It depends on what you consider an ‘error http response’ which depends on how the service works. 
If the serviec returns an error HTTP status axios will throw an error, and the client must construct a suitable PolicystoreError in the catch block. 
If it returns a 2xx but there are some conditions in the data returned that make the client decide the response is not good, then the client can throw the PolicystoreError itself after checking for that condition in the try block.

**Comment by jordibuj** [comment]
I’m not sure you have to develop specific tests for the policystore client as it is a very straightforward piece of code, and the service is already thoroughly tested. 
I’m fine not having any specific tests for the credential issuer client, the dddc graphql client or the bcnnow login client, and the same holds for the policystore client.
The critical point to test in this app is the 'business logic’ which is implemented in the Redux modules.
So it is just fine having a Policystore client without tests, if it exposes the functions that are required by the Redux module, returning the data objects that the module needs, and throwing a suitable error otherwise.

**Comment by smulube** [comment]
Thanks for the review comments @jordibuj - closing this PR for two more focussed PRs containing the cleaned up versions of the above code.

**Pull request closed without merging**